libp2p/protocols/pubsub/gossipsub/scoring.nim

```diff
@@ -250,43 +250,46 @@ proc punishInvalidMessage*(g: GossipSub, peer: PubSubPeer, msg: Message) {.async
         await g.disconnectPeer(peer)
         raise newException(PeerRateLimitError, "Peer disconnected because it's above rate limit.")
 
+  let topic = msg.topic
+  # TODO: REVIEW
+  if topic notin g.topics:
+    return
 
-  for tt in msg.topicIds:
-    let t = tt
-    if t notin g.topics:
-      continue
-
-    let tt = t
-    # update stats
-    g.withPeerStats(peer.peerId) do (stats: var PeerStats):
-      stats.topicInfos.mgetOrPut(tt, TopicInfo()).invalidMessageDeliveries += 1
+  # update stats
+  g.withPeerStats(peer.peerId) do(stats: var PeerStats):
+    stats.topicInfos.mgetOrPut(topic, TopicInfo()).invalidMessageDeliveries += 1
 
 proc addCapped*[T](stat: var T, diff, cap: T) =
   stat += min(diff, cap - stat)
 
 proc rewardDelivered*(
-    g: GossipSub, peer: PubSubPeer, topics: openArray[string], first: bool, delay = ZeroDuration) =
-  for tt in topics:
-    let t = tt
-    if t notin g.topics:
-      continue
-
-    let tt = t
-    let topicParams = g.topicParams.mgetOrPut(t, TopicParams.init())
-                                            # if in mesh add more delivery score
-
-    if delay > topicParams.meshMessageDeliveriesWindow:
-      # Too old
-      continue
-
-    g.withPeerStats(peer.peerId) do (stats: var PeerStats):
-      stats.topicInfos.withValue(tt, tstats):
-        if first:
-          tstats[].firstMessageDeliveries.addCapped(
-            1, topicParams.firstMessageDeliveriesCap)
-
-        if tstats[].inMesh:
-          tstats[].meshMessageDeliveries.addCapped(
-            1, topicParams.meshMessageDeliveriesCap)
-      do: # make sure we don't loose this information
-        stats.topicInfos[tt] = TopicInfo(meshMessageDeliveries: 1)
+    g: GossipSub,
+    peer: PubSubPeer,
+    topic: string,
+    first: bool,
+    delay = ZeroDuration,
+) =
+  let t = topic
+  if t notin g.topics:
+    return
+
+  let topic = t
```

Comment:
These two assignations `let t = topic` and then `let topic = t` confuse me so much, do you have any idea why it was this way?

Reply:
No, and when I simplified it it stopped working 😂😂

Reply:
So I dug a bit into it and there's two things related:
- `let t = topic` : https://github.com/vacp2p/nim-libp2p/issues/581
- `let topic = t` : https://github.com/vacp2p/nim-libp2p/commit/ce42674d80222e5dc6f9c779cef128005d696bf9
Both were issues raised with nim1.4, I can understand why the first hasn't been fixed (or even if it's fixable), but the second seems a bit odd. What happens if you just remove the second assignation?

Reply:
I ended up removing the two "unnecessary" assigments and tests seemed to pass locally. I did that once previously and tests failed, so... I don't know. Let's see what the CI says.

Reply:
Passed, amazing. All good then.